src/Stratis.Features.SQLiteWalletRepository/Tables/HDAddress.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Collections.Generic;
+using SQLite;
+
+namespace Stratis.Features.SQLiteWalletRepository.Tables
+{
+    internal class HDAddress
+    {
+        public int WalletId { get; set; }
+        public int AccountIndex { get; set; }
+        public int AddressType { get; set; }
+        public int AddressIndex { get; set; }
+        public string ScriptPubKey { get; set; }
+        public string PubKey { get; set; }
```

Comment:
This is a bit of an implementation wart. It would be better to only have the concept of a scriptPubKey, as a P2PK script is just a scriptPubKey of a different template format.

However, it is commonplace for wallet implementations and block explorers to (incorrectly!) treat P2PK scriptPubKeys as having an address. They do this by treating the P2PK pubKey as P2PKH and using the P2PKH address instead.

So having this field means bolting an additional script onto everything, as well as checking against the two scripts for every incoming transaction. It is typically only the mining code that uses the pubKey. In future it may make sense to move away from it entirely and only add P2PK addresses as-needed for mining, instead of treating every address with this duality.

Reply:
You are correct. In fact, the `PubKey` field is not being used and not being matched with anything. The only reason I am including it is to have something one can derive the original public key of the address from. I guess I'm just clinging onto information that would be lost otherwise and may have some future application.